### PR TITLE
Better document Storage API

### PR DESCRIPTION
**Pull request by mziccard**
This follows from #351

### CODE REVIEW

File: gcloud-java-storage/src/main/java/com/google/gcloud/storage/BlobInfo.java [hunk 1]
```diff
@@ -158,11 +173,20 @@ Builder componentCount(Integer componentCount) {
       return this;
     }
 
+    /**
+     * Sets the blob's data cache control.
```
Comment (aozarov) [inline gcloud-java-storage/src/main/java/com/google/gcloud/storage/BlobInfo.java]: Would be nice to include the reference to the valid structure.
e.g. in this case refer to https://tools.ietf.org/html/rfc7234#section-5.2.
See [this](https://cloud.google.com/storage/docs/json_api/v1/objects/insert) for example that includes links for other settings such as content-Type, Content-Disposition,...

File: gcloud-java-storage/src/main/java/com/google/gcloud/storage/BlobInfo.java [hunk 2]
```diff
@@ -259,98 +296,177 @@ private BlobInfo(Builder builder) {
...
 
+  /**
+   * Returns blob's metadata.
+   */
   public Map<String, String> metadata() {
     return metadata == null || Data.isNull(metadata) ? null : Collections.unmodifiableMap(metadata);
   }
 
+  /**
+   * Returns blob's data generation.
```
Comment (aozarov) [inline gcloud-java-storage/src/main/java/com/google/gcloud/storage/BlobInfo.java]: Did you consider copy/paste from here https://cloud.google.com/storage/docs/json_api/v1/objects#resource.
I think some of the descriptions there are nice.

File: gcloud-java-storage/src/main/java/com/google/gcloud/storage/Storage.java [hunk 3]
```diff
@@ -441,14 +579,23 @@ private BucketListOption(StorageRpc.Option option, Object value) {
...
 
+    /**
+     * Returns an option to specify the page token from which to start listing buckets.
+     */
     public static BucketListOption startPageToken(String pageToken) {
       return new BucketListOption(StorageRpc.Option.PAGE_TOKEN, pageToken);
     }
 
+    /**
+     * Returns an option to set a prefix for listed buckets names.
```
Comment (aozarov) [inline gcloud-java-storage/src/main/java/com/google/gcloud/storage/Storage.java]: Returns an option to set a prefix to filter results to buckets whose names begin with this prefix.

Description taken from [here](https://cloud.google.com/storage/docs/json_api/v1/buckets/list).

File: gcloud-java-storage/src/main/java/com/google/gcloud/storage/Storage.java [hunk 4]
```diff
@@ -474,18 +624,30 @@ private BlobListOption(StorageRpc.Option option, Object value) {
...
 
+    /**
+     * Returns an option to specify the page token from which to start listing blobs.
+     */
     public static BlobListOption startPageToken(String pageToken) {
       return new BlobListOption(StorageRpc.Option.PAGE_TOKEN, pageToken);
     }
 
+    /**
+     * Returns an option to set a prefix for listed blobs names.
```
Comment (aozarov) [inline gcloud-java-storage/src/main/java/com/google/gcloud/storage/Storage.java]: Returns an option to set a prefix to filter results to blobs whose names begin with this prefix.

**Comment by aozarov** [comment]
Do you think BucketInfo is already well documented (compared to https://cloud.google.com/storage/docs/json_api/v1/buckets#resource)?

**Comment by aozarov** [comment]
Ignore my last comment. Just noticed that you added javadoc for `BucketInfo` in #355.

**Comment by mziccard** [comment]
I fixed the comments and integrated javadoc

**Pull request merged**